File: 162. Find Peak Element.swift
class Solution {
    func findPeakElement(_ nums: [Int]) -> Int {
        guard nums.count > 0 else {
            return -1
        }
        
        if nums.count == 1 {
            return 0
        }
        
        var left = nums.startIndex
        var right = nums.endIndex - 1
        
        while left < right {
            let mid = left + (right - left) / 2
            
            if nums[mid] < nums[mid+1] {
                left = mid + 1
            } else if nums[mid] > nums[mid+1] {
                right = mid
            }
        }
        
        return left
    }
}

File: 162.Find Peak Element.cpp
class Solution {
public:
    int findPeakElement(vector<int>& nums) {
        if (nums.size() <= 1 || nums[0] > nums[1])
            return 0;
            
        for (int i=1; i<nums.size()-1; i++) {
            if (nums[i-1] < nums[i] && nums[i] > nums[i+1])
                return i;
        }
        
        return nums.size()-1;
    }
};
